
package com.caredear;

enum PasswordType{
    MODIFY = 0;
    FORGET = 1;
}

message PasswordManagerRequest{
    required PasswordType type = 1; // change ? forget ? etc...(currently don't use this)
    required uint64  caredear_id = 2; // the user's CID
    required string  new_passwd = 3;
    optional string  old_passwd = 4; // If contained this, user intent to tell me he verify old passwd
    optional string  cur_token = 5; // added as for BUG-6178
}

message PasswordManagerResponse{
    required int32 result_code = 1; // CDS_XXX code
    optional string extra_msg = 2; // further error msg if it has
}
